index.js

```diff
@@ -55,6 +62,11 @@
   }
   extend(Nothing, Maybe);
```

Comment:
Interesting! I don't fully understand that code upon first reading. I'll revisit it this weekend to work out exactly what it does. :)